File: 2sem/lab5/lab5/BaseString.cpp
#include "BaseString.h"
#include <string>

BaseString::BaseString() {  // default constructor
	_string = "";
}

BaseString::BaseString(std::string str) {   // constructor with parameter
	_string = str;
}

int BaseString::GetLength() {  // getter or length
	return _string.length();
}

File: 2sem/lab5/lab5_sharp/BaseString.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace lab5_sharp
{
    class BaseString
    {
        public string String { get; protected set; } 

        public BaseString() { }  // default counstructor
        public BaseString(string str)  // constructor with parameter
        {
            String = str;
        }

        public int GetLength()  // getter of length
        {
            return String.Length;
        }
    }
}
